great_expectations/datasource/fluent/interfaces.py

```diff
@@ -373,6 +373,7 @@ class Datasource(
     assets: MutableMapping[str, _DataAssetT] = {}
 
     # private attrs
+    _data_context = pydantic.PrivateAttr()
```

Comment:
`non-blocking`
Sorry I've already forgotten why we need this. But could we just store whatever the needed attribute/object is from the datacontext instead of the whole context?